File: 3349. Adjacent Increasing Subarrays Detection I.cpp
class Solution {
public:
    bool hasIncreasingSubarrays(vector<int>& nums, int k) {
        int n = nums.size();
    
    for (int i = 0; i <= n - 2 * k; ++i) {
        
        bool firstIncreasing = true;
        for (int j = i; j < i + k - 1; ++j) {
            if (nums[j] >= nums[j + 1]) {
                firstIncreasing = false;
                break;
            }
        }

        bool secondIncreasing = true;
        for (int j = i + k; j < i + 2 * k - 1; ++j) {
            if (nums[j] >= nums[j + 1]) {
                secondIncreasing = false;
                break;
            }
        }

        if (firstIncreasing && secondIncreasing) {
            return true;
        }
    }

    return false;
    }
};

File: 3349. Adjacent Increasing Subarrays Detection I.java
class Solution {
    public boolean hasIncreasingSubarrays(List<Integer> nums, int k) {
        int n = nums.size();
        boolean[] hasSub = new boolean[n];
        int c = 1;
        if(k == 1) hasSub[0] = true;
        for(int i = 1; i < n; i++){
            if(nums.get(i) > nums.get(i - 1)) c++;
            else c = 1;

            if(c >= k) hasSub[i] = true;
        }

        for(int i = 0; i < n; i++){
            // System.out.println(hasSub[i]);
            if(hasSub[i] && i + k < n && hasSub[i + k]) return true;
        }

        return false;
    }
}